subprojects/core/src/main/java/org/gradle/api/internal/plugins/ApplyPluginBuildOperationType.java

```diff
@@ -31,30 +30,33 @@
     @UsedByScanPlugin
     public interface Details {
 
+        /**
+         * The fully qualified plugin ID, if known.
+         */
         @Nullable
         String getPluginId();
 
+        /**
+         * The fully qualified class name of the Plugin implementation.
+         */
         String getClassName();
 
-    }
-
-    static class DetailsImpl implements Details {
-        private PluginId pluginId;
-        private String className;
-
-        DetailsImpl(@Nullable PluginId pluginId, String className) {
-            this.pluginId = pluginId;
-            this.className = className;
-        }
+        /**
+         * The target of the plugin.
+         * One of "gradle", "settings", "project" or null.
+         */
+        String getTargetType();
 
+        /**
+         * If the target is a project, its path.
+         */
         @Nullable
-        public String getPluginId() {
-            return pluginId.getId();
-        }
+        String getTargetPath();
 
-        public String getClassName() {
-            return className;
-        }
+        /**
+         * The build path of the target.
+         */
+        String getBuildPath();
     }
```

Comment:
Can we also add an API that allows us to populate [PluginCodeSourceType_1](https://github.com/gradle/dotcom/blob/65eb258b54f93e1563654e54e43a7b0b64051fc1/scans-event-model/src/main/java/com/gradle/scan/eventmodel/plugin/PluginCodeSourceType_1.java#L9-L9)?